packages/gatsby/src/redux/actions/public.js

```diff
@@ -1478,4 +1479,50 @@ actions.unstable_createNodeManifest = (
   }
 }
 
+/**
+ * Stores request headers for a given domain to be later used when making requests for Image CDN (and potentially other features).
+ *
+ * @param {Object} $0
+ * @param {string} $0.domain The domain to store the headers for.
+ * @param {Object} $0.headers The headers to store.
+ */
+actions.setRequestHeaders = ({ domain, headers }, plugin: Plugin) => {
+  const noHeaders = typeof headers !== `object`
```

Comment:
This check is not sufficient if you really need to know if it's an object because `typeof []` will also return object. If you really want to check if it's an Object you need to do something like `Object.prototype.toString.call(obj) === '[object Object]';`